src/jit/gentree.h

```diff
@@ -1747,6 +1753,15 @@ struct GenTree
     // sets "*pIsEntire" to true if this assignment writes the full width of the local.
     bool DefinesLocalAddr(Compiler* comp, unsigned width, GenTreeLclVarCommon** pLclVarTree, bool* pIsEntire);
 
+    regNumber RegAtDef() const
+    {
+        return GetRawRegNum();
+    }
+    regNumber RegAtUse() const
+    {
+        return isContained() ? REG_NA : GetRawRegNum();
+    }
+
```

Comment:
I think it would be better in `CodeGen`, and/or the methods that deal with containedness need to be clearly delineated, and need to have clear preconditions, with associated asserts validating them, to ensure that they are being called when the containedness is valid.

---
In reply to: [85577137](https://github.com/dotnet/coreclr/pull/7876#discussion_r85577137) [](ancestors = 85577137)